src/Routers/FunctionsRouter.js

```diff
@@ -133,52 +114,34 @@ export class FunctionsRouter extends PromiseRouter {
       context: req.info.context,
     };
 
-    return new Promise(function (resolve, reject) {
-      const userString = req.auth && req.auth.user ? req.auth.user.id : undefined;
-      const cleanInput = logger.truncateLogMessage(JSON.stringify(params));
-      const { success, error } = FunctionsRouter.createResponseObject(
-        result => {
-          try {
-            const cleanResult = logger.truncateLogMessage(JSON.stringify(result.response.result));
-            logger.info(
-              `Ran cloud function ${functionName} for user ${userString} with:\n  Input: ${cleanInput}\n  Result: ${cleanResult}`,
-              {
-                functionName,
-                params,
-                user: userString,
-              }
-            );
-            resolve(result);
-          } catch (e) {
-            reject(e);
-          }
-        },
-        error => {
-          try {
-            logger.error(
-              `Failed running cloud function ${functionName} for user ${userString} with:\n  Input: ${cleanInput}\n  Error: ` +
-                JSON.stringify(error),
-              {
-                functionName,
-                error,
-                params,
-                user: userString,
-              }
-            );
-            reject(error);
-          } catch (e) {
-            reject(e);
-          }
+    const userString = req.auth?.user?.id;
+    const cleanInput = logger.truncateLogMessage(JSON.stringify(params));
+    try {
+      await maybeRunValidator(request, functionName, req.auth);
+      const response = await theFunction(request);
+      const result = Parse._encode(response);
+      const cleanResult = logger.truncateLogMessage(JSON.stringify(result));
+      logger.info(
+        `Ran cloud function ${functionName} for user ${userString} with:\n  Input: ${cleanInput}\n  Result: ${cleanResult}`,
+        {
+          functionName,
+          params,
+          user: userString,
         }
       );
-      return Promise.resolve()
-        .then(() => {
-          return triggers.maybeRunValidator(request, functionName, req.auth);
-        })
-        .then(() => {
-          return theFunction(request);
-        })
-        .then(success, error);
-    });
+      return {
+        response: {
+          result,
+        },
+      };
+    } catch (e) {
+      const error = resolveError(e);
+      logger.error(
+        `Failed running cloud function ${functionName} for user ${userString} with:\n  Input: ${cleanInput}\n  Error: ${JSON.stringify(
+          error
+        )}\n  Stack: ${error.stack}\n`
```

Comment:
I should note that this `Stack: ${error.stack}` is the only addition in this PR @mtrezza

Reply:
looks good